### PR TITLE
Add add_service_for_host endpoint to API

### PR DESCRIPTION
**Pull request by remyj38**
DO NOT DELETE THIS TEXT

#### Please note

> Please read this information carefully. You can run `./scripts/pre-commit.php` to check your code before submitting.

- [x] Have you followed our [code guidelines?](http://docs.librenms.org/Developing/Code-Guidelines/)

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`

This pull request add the add_service_for_device API endpoint.

### CODE REVIEW

File: html/api_v0.php [hunk 1]
```diff
@@ -186,6 +186,7 @@ function () use ($app) {
                     '/services',
                     function () use ($app) {
                         $app->get('/:hostname', 'authToken', 'list_services')->name('get_service_for_host');
+                        $app->post('/:hostname', 'authToken', 'add_service_for_host')->name('add_service_for_host');
```
Comment (laf) [inline html/api_v0.php]: I'd change `add_service_for_host` to just `add_service`.

File: html/includes/api_functions.inc.php [hunk 2]
```diff
@@ -1513,3 +1513,52 @@ function validate_column_list($columns, $tableName)
         $app->stop();
     }
 }
+
+function add_service_for_host()
+{
+    check_is_admin();
```
Comment (laf) [inline html/includes/api_functions.inc.php]: We should also allow normal users to add these if they are allowed to access the device.

You can use `check_device_permission($device_id);` once you have the device id below.

File: html/api_v0.php (hunk 1 above)
Comment (remyj38) [reply html/api_v0.php]: It's already defined in https://github.com/librenms/librenms/blob/master/includes/services.inc.php#L37 and used in the function

File: html/includes/api_functions.inc.php (hunk 2 above)
Comment (remyj38) [reply html/includes/api_functions.inc.php]: Changed in 7344482

File: html/api_v0.php (hunk 1 above)
Comment (laf) [reply html/api_v0.php]: Good shout :)

**Comment by laf** [comment]
Thanks very much for contributing this @remyj38 

image

**Pull request merged**